client.go

```diff
@@ -619,3 +620,58 @@ func (c *Client) watch(ctx context.Context, url string) (*watcher, error) {
 	w := &watcher{resp.Body}
 	return w, nil
 }
+
+// JSONEvent is an event for ThirdPartyResources that do not support protobuf over the wire, only json
+type JSONEvent struct {
+	Type   string      `json:"type,omitempty"`
+	Object interface{} `json:"object,omitempty"`
+}
+
+type jsonWatcher struct {
+	r io.ReadCloser
+	d *json.Decoder
+}
+
+func (w *jsonWatcher) Close() error {
+	return w.r.Close()
+}
+
+// Decode the next event from a watch stream of type application/json.
+func (w *jsonWatcher) nextJSON(object interface{}) (*JSONEvent, error) {
+
+	var event JSONEvent
+	if err := w.d.Decode(&event); err != nil {
+		return nil, err
+	}
+
+	return &event, nil
+}
+
+func (c *Client) watchJSON(ctx context.Context, url string) (*jsonWatcher, error) {
```

Comment:
Instead of duplicating this code, we should probably just let watcher take a codec.

Reply:
I tried that first, but the parseUnknown and event object are pretty hard to unify, which led to a big switch depending on the codec. If you think that's nicer I'm fine trying to do that instead.